rhealpixdggs/rhp_wrappers.py

```diff
@@ -0,0 +1,537 @@
+    cell = WGS84_003.cell(suid)
+    verts = tuple(cell.vertices(plane=plane))
+
+    # rhealpix coordinates come out natively as lng/lat, h3 ones as lat/lng
+    # Neither has the repeated vertex that geo_json wants so it's inserted here when needed
+    if not geo_json:
+        # lng/lat -> lat/lng to make it consistent with h3
+        verts = tuple(v[::-1] for v in verts)
+    else:
+        # last point same as first
+        verts += (verts[0],)
+
+    return verts
+
+
+def rhp_get_resolution(rhpindex: str) -> int:
+    """
+    Returns the resolution of a given cell index (or None if invalid).
+    """
+    if not rhp_is_valid(rhpindex):
+        return None
+
+    return len(rhpindex) - 1
+
+
+def rhp_get_base_cell(rhpindex: str) -> str:
+    """
+    Returns the resolution 0 cell id of a given cell index (or None if invalid).
+    """
+    if not rhp_is_valid(rhpindex):
+        return None
+
+    return rhpindex[0]
+
+
+def rhp_is_valid(rhpindex: str) -> bool:
+    # TODO: call this function in those that include address-based functionality
+    #       i.e. anything that accepts a string 'rhpindex' as an argument
+    """
+    Checks if the given cell address is valid within the DGGS
+
+    TODO: give the option to select another predefined DGGS, or pass in a custom one
+    TODO: give the option to set n to something other than 3
+    """
+    # Empty strings are invalid
+    if rhpindex is None or len(rhpindex) == 0:
+        return False
+
+    # Addresses that don't start with the resolution 0 face are invalid
+    if rhpindex[0] not in CELLS0:
+        return False
+
+    # Addresses that have digits out of range are invalid
+    num_subcells = WGS84_003.N_side**2
+    for d in rhpindex[1:]:
+        if not d.isdigit() or (int(d) >= num_subcells):
+            return False
+
+    # Passed all checks - must be the real thing
+    return True
+
+
+def cell_area(
+    rhpindex: str, unit: Literal["km^2", "m^2"] = "km^2", plane=True
+) -> float:
+    """
+    Returns the area of a cell in the requested unit (or None if rhpindex is invalid).
+
+    TODO: investigate use case where unit is 'rads^2'
+    """
+    if not rhp_is_valid(rhpindex):
+        return None
+
+    # Grab cell area in native unit (m^2)
+    suid = [int(d) if d.isdigit() else d for d in rhpindex]
+    cell = WGS84_003.cell(suid)
+    area = cell.area(plane=plane)
+
+    # Scale area if needed
+    if unit == "km^2":
+        area = area / 10**6
+
+    return area
+
+
+def cell_ring(rhpindex: str, k: int = 1, verbose=True) -> list[str]:
+    """
+    Returns the ring of cell indices around rhpindex at distance k, or None if rhpindex
+    is invalid.
+
+    Also returns None if k < 0.
+
+    Returns [ rhpindex ] if k == 0 (by convention).
+
+    Returns the four neighbouring faces at resolution 0 if k > 0 and cell resolution is 0
+    (by convention).
+    """
+    if verbose:
+        print(str.format(CELL_RING_WARNING, "cell"))
```

Comment:
Should this rather use the built-in warnings module? https://docs.python.org/3/library/warnings.html#warnings.warn

Reply:
You're absolutely right - I've changed that.